src/sas/qtgui/MainWindow/PackageGatherer.py

```diff
@@ -0,0 +1,228 @@
+import sys, logging, subprocess, pkg_resources, json
+
+
+class PackageGatherer:
+    """
+    A class used to gather packages/modules  used by SasView and their current installed version
+
+    Methods
+    -------
+    log_installed_modules
+        Log version number of locally installed python packages
+
+    log_imported_modules
+        Log version number of python packages imported in this instance of SasView.
+
+    get_imported_modules
+        Get a dictionary of imported module version numbers
+
+    remove_duplicate_modules
+        Strip duplicate instances of each module
+
+    format_unattainable_modules_list
+        Format module names in the unattainable_modules list
+    """
+
+
+    def log_installed_modules(self):
+        """ Log version number of locally installed python packages
+
+        Use pip list to create a dictionary of installed modules as the keys, with their respective version numbers
+        as the values. Only packages available through pip will be included.
+
+        Returns
+        -------
+        None
+        """
+        installed_modules = {'python': sys.version}
+
+        # Get python modules installed locally
+        installed_modules_json = json.loads(subprocess.check_output("pip list -l --format=json"))
+        for mod in installed_modules_json:
+            installed_modules[mod['name']] = mod['version']
+
+        logging.info(f"Installed modules\n"
+                     f"{installed_modules}")
+
+
+    def log_imported_modules(self):
+        """ Log version number of python packages imported in this instance of SasView.
+
+        Use the get_imported_modules method to to create a dictionary of installed modules as the keys, with their
+        respective version numbers as the values. There may be some packages whose version number is unattainable.
+
+        Returns
+        -------
+        None
+        """
+        imported_modules = self.get_imported_modules()
+
+        logging.info(f"Imported modules\n"
+                     f"{imported_modules}")
+
+
+    def get_imported_modules(self):
+        """ Get a dictionary of imported module version numbers
+
+        Use a variaty of method, for example a module.version call, to attempt to get the module version of each
+        module that has been imported in this instance of running SasView. The sys.modules command lists the
+        imported modules. A list of modules who's version number cannot be found is also included.
+
+        Returns
+        -------
+        module_versions_dict : dict
+            A dictionary with the module names as the key, with their respective version numbers as the value.
+        """
+
+        module_versions_dict = {'python': sys.version}
+        unattainable_modules = []
+
+        for mod in sys.modules.keys():
+
+            # Not a built in python module, which has no version, only the python version
+            if mod not in sys.builtin_module_names:
```

Comment:
Hi paul, I see how deep nested try except blocks are ahard to read, so i've reformatted the code to avoid this. 
I've also used the python library to generate a list of standard modules as you suggested, which works well.